plugins/TelescopeControl/src/common/Socket.cpp

```diff
@@ -40,7 +40,7 @@ long long int GetNow(void)
 		qint64 t;
 	} tmp;
 	GetSystemTimeAsFileTime(&tmp.file_time);
-	t = (tmp.t/10) - 86400000000LL*(369*365+89);
+	t = (tmp.t/10) - 86400000000LL*(369*365+89); // TODO: Explain this magic please!
```

Comment:
The constant is used to convert from NT/NTFS time epoch (1601-01-01 0:00:00 UTC) in μs to UNIX epoch (1970-01-01 0:00:00 UTC) in μs. The difference is:

24 hour/day × 3600 s/hour × ((369×365+89) day).

Here 369=1970-1601, and 89 is the number of leap years in the interval. You can verify that 369×365+89=134774 is equal to the output of this:

```
python -c "from datetime import date as d; print (d(1970,1,1) - d(1601,1,1)).days"
```
The division by 10 in the minuend is to convert from 100-nanosecond intervals to microseconds.

Reply:
I suspected something around this, but was unaware of the 1601 epoch for NTFS. 89 leap years, right, thanks!